// @livekit/components-react/dist/components-CG43NVTo.mjs
import * as e from "react";
import { useState as x, useRef as I, useEffect as P, useMemo as ge } from "react";
import { u as ve, a as K, b as Ee, c as pe, d as we, e as ke, f as ee, g as Me, h as Re, i as ye, j as be, k as Se, l as te, m as Ce, n as Ie, o as Pe, p as Te, q as xe, r as Ae, s as ae, t as Ne, v as Le } from "./hooks-aMI94FOv.mjs";
import { a as y, c as O } from "./room-DfA7knPM.mjs";
import { RoomEvent as ze, Track as M, ConnectionQuality as V, RemoteTrackPublication as H, RemoteAudioTrack as Ve, ConnectionState as F } from "livekit-client";
import { a9 as _, aa as ne, w as Fe, f as re, c as Ze, ab as He, X as Be, Y as je, a1 as z, l as A, ac as ce, ad as L, ae as Oe, i as N, B as _e, af as le, ag as $, j as $e, ah as We, m as qe, p as ie, ai as De, aj as Ge, ak as Qe, al as Ue, am as Xe } from "./contexts-568EHGLp.mjs";
const ea = /* @__PURE__ */ e.forwardRef(
  function(n, a) {
    const { buttonProps: r } = ve(n);
    return /* @__PURE__ */ e.createElement("button", { ref: a, ...r }, n.children);
  }
), ta = /* @__PURE__ */ e.forwardRef(
  function({ room: n, ...a }, r) {
    const c = K(n);
    return /* @__PURE__ */ e.createElement("div", { ref: r, ...a }, c);
  }
), aa = /* @__PURE__ */ e.forwardRef(
  function(n, a) {
    const { mergedProps: r } = Ee({ props: n });
    return /* @__PURE__ */ e.createElement("button", { ref: a, ...r }, n.children);
  }
), na = /* @__PURE__ */ e.forwardRef(
  function(n, a) {
    const { buttonProps: r } = pe(n);
    return /* @__PURE__ */ e.createElement("button", { ref: a, ...r }, n.children);
  }
), Ye = (t) => /* @__PURE__ */ e.createElement("svg", { xmlns: "http://www.w3.org/2000/svg", width: 16, height: 16, fill: "currentColor", ...t }, /* @__PURE__ */ e.createElement("path", { d: "M1.354.646a.5.5 0 1 0-.708.708l14 14a.5.5 0 0 0 .708-.708L11 10.293V4.5A1.5 1.5 0 0 0 9.5 3H3.707zM0 4.5a1.5 1.5 0 0 1 .943-1.393l9.532 9.533c-.262.224-.603.36-.975.36h-8A1.5 1.5 0 0 1 0 11.5z" }), /* @__PURE__ */ e.createElement("path", { d: "m15.2 3.6-2.8 2.1a1 1 0 0 0-.4.8v3a1 1 0 0 0 .4.8l2.8 2.1a.5.5 0 0 0 .8-.4V4a.5.5 0 0 0-.8-.4z" })), Je = (t) => /* @__PURE__ */ e.createElement("svg", { xmlns: "http://www.w3.org/2000/svg", width: 16, height: 16, fill: "currentColor", ...t }, /* @__PURE__ */ e.createElement("path", { d: "M0 4.5A1.5 1.5 0 0 1 1.5 3h8A1.5 1.5 0 0 1 11 4.5v7A1.5 1.5 0 0 1 9.5 13h-8A1.5 1.5 0 0 1 0 11.5zM15.2 3.6l-2.8 2.1a1 1 0 0 0-.4.8v3a1 1 0 0 0 .4.8l2.8 2.1a.5.5 0 0 0 .8-.4V4a.5.5 0 0 0-.8-.4z" })), ra = (t) => /* @__PURE__ */ e.createElement("svg", { xmlns: "http://www.w3.org/2000/svg", width: 16, height: 16, viewBox: "0 0 24 24", ...t }, /* @__PURE__ */ e.createElement(
  "path",
  {
    fill: "#FFF",
    d: "M4.99 3.99a1 1 0 0 0-.697 1.717L10.586 12l-6.293 6.293a1 1 0 1 0 1.414 1.414L12 13.414l6.293 6.293a1 1 0 1 0 1.414-1.414L13.414 12l6.293-6.293a1 1 0 0 0-.727-1.717 1 1 0 0 0-.687.303L12 10.586 5.707 4.293a1 1 0 0 0-.717-.303z"
  }
)), ca = (t) => /* @__PURE__ */ e.createElement("svg", { xmlns: "http://www.w3.org/2000/svg", width: 16, height: 18, fill: "none", ...t }, /* @__PURE__ */ e.createElement(
  "path",
  {
    fill: "currentColor",
    fillRule: "evenodd",
    d: "M0 2.75A2.75 2.75 0 0 1 2.75 0h10.5A2.75 2.75 0 0 1 16 2.75v13.594a.75.75 0 0 1-1.234.572l-3.691-3.12a1.25 1.25 0 0 0-.807-.296H2.75A2.75 2.75 0 0 1 0 10.75v-8ZM2.75 1.5c-.69 0-1.25.56-1.25 1.25v8c0 .69.56 1.25 1.25 1.25h7.518c.65 0 1.279.23 1.775.65l2.457 2.077V2.75c0-.69-.56-1.25-1.25-1.25H2.75Z",
    clipRule: "evenodd"
  }
), /* @__PURE__ */ e.createElement(
  "path",
  {
    fill: "currentColor",
    fillRule: "evenodd",
    d: "M3 4.5a.5.5 0 0 1 .5-.5h9a.5.5 0 0 1 0 1h-9a.5.5 0 0 1-.5-.5Zm0 2a.5.5 0 0 1 .5-.5h9a.5.5 0 0 1 0 1h-9a.5.5 0 0 1-.5-.5Zm0 2a.5.5 0 0 1 .5-.5h5a.5.5 0 0 1 0 1h-5a.5.5 0 0 1-.5-.5Z",
    clipRule: "evenodd"
  }
)), D = (t) => /* @__PURE__ */ e.createElement("svg", { xmlns: "http://www.w3.org/2000/svg", width: 16, height: 16, fill: "none", ...t }, /* @__PURE__ */ e.createElement(
  "path",
  {
    fill: "currentcolor",
    fillRule: "evenodd",
    d: "M5.293 2.293a1 1 0 0 1 1.414 0l4.823 4.823a1.25 1.25 0 0 1 0 1.768l-4.823 4.823a1 1 0 0 1-1.414-1.414L9.586 8 5.293 3.707a1 1 0 0 1 0-1.414z",
    clipRule: "evenodd"
  }
)), Ke = (t) => /* @__PURE__ */ e.createElement("svg", { xmlns: "http://www.w3.org/2000/svg", width: 16, height: 16, fill: "none", ...t }, /* @__PURE__ */ e.createElement("g", { stroke: "currentColor", strokeLinecap: "round", strokeLinejoin: "round", strokeWidth: 1.5 }, /* @__PURE__ */ e.createElement("path", { d: "M10 1.75h4.25m0 0V6m0-4.25L9 7M6 14.25H1.75m0 0V10m0 4.25L7 9" }))), la = (t) => /* @__PURE__ */ e.createElement("svg", { xmlns: "http://www.w3.org/2000/svg", width: 16, height: 16, fill: "none", ...t }, /* @__PURE__ */ e.createElement(
  "path",
  {
    fill: "currentcolor",
    fillRule: "evenodd",
    d: "M8.961.894C8.875-.298 7.125-.298 7.04.894c-.066.912-1.246 1.228-1.76.472-.67-.99-2.186-.115-1.664.96.399.824-.465 1.688-1.288 1.289-1.076-.522-1.95.994-.961 1.665.756.513.44 1.693-.472 1.759-1.192.086-1.192 1.836 0 1.922.912.066 1.228 1.246.472 1.76-.99.67-.115 2.186.96 1.664.824-.399 1.688.465 1.289 1.288-.522 1.076.994 1.95 1.665.961.513-.756 1.693-.44 1.759.472.086 1.192 1.836 1.192 1.922 0 .066-.912 1.246-1.228 1.76-.472.67.99 2.186.115 1.664-.96-.399-.824.465-1.688 1.288-1.289 1.076.522 1.95-.994.961-1.665-.756-.513-.44-1.693.472-1.759 1.192-.086 1.192-1.836 0-1.922-.912-.066-1.228-1.246-.472-1.76.99-.67.115-2.186-.96-1.664-.824.399-1.688-.465-1.289-1.288.522-1.076-.994-1.95-1.665-.961-.513.756-1.693.44-1.759-.472ZM8 13A5 5 0 1 0 8 3a5 5 0 0 0 0 10Z",
    clipRule: "evenodd"
  }
)), ia = (t) => /* @__PURE__ */ e.createElement("svg", { xmlns: "http://www.w3.org/2000/svg", width: 16, height: 16, fill: "none", ...t }, /* @__PURE__ */ e.createElement(
  "path",
  {
    fill: "currentColor",
    fillRule: "evenodd",
    d: "M2 2.75A2.75 2.75 0 0 1 4.75 0h6.5A2.75 2.75 0 0 1 14 2.75v10.5A2.75 2.75 0 0 1 11.25 16h-6.5A2.75 2.75 0 0 1 2 13.25v-.5a.75.75 0 0 1 1.5 0v.5c0 .69.56 1.25 1.25 1.25h6.5c.69 0 1.25-.56 1.25-1.25V2.75c0-.69-.56-1.25-1.25-1.25h-6.5c-.69 0-1.25.56-1.25 1.25v.5a.75.75 0 0 1-1.5 0v-.5Z",
    clipRule: "evenodd"
  }
), /* @__PURE__ */ e.createElement(
  "path",
  {
    fill: "currentColor",
    fillRule: "evenodd",
    d: "M8.78 7.47a.75.75 0 0 1 0 1.06l-2.25 2.25a.75.75 0 1 1-1.06-1.06l.97-.97H1.75a.75.75 0 0 1 0-1.5h4.69l-.97-.97a.75.75 0 0 1 1.06-1.06l2.25 2.25Z",
    clipRule: "evenodd"
  }
)), et = (t) => /* @__PURE__ */ e.createElement("svg", { xmlns: "http://www.w3.org/2000/svg", width: 16, height: 16, fill: "none", ...t }, /* @__PURE__ */ e.createElement(
  "path",
  {
    fill: "currentcolor",
    fillRule: "evenodd",
    d: "M4 6.104V4a4 4 0 1 1 8 0v2.104c1.154.326 2 1.387 2 2.646v4.5A2.75 2.75 0 0 1 11.25 16h-6.5A2.75 2.75 0 0 1 2 13.25v-4.5c0-1.259.846-2.32 2-2.646ZM5.5 4a2.5 2.5 0 0 1 5 0v2h-5V4Z",
    clipRule: "evenodd"
  }
)), tt = (t) => /* @__PURE__ */ e.createElement("svg", { xmlns: "http://www.w3.org/2000/svg", width: 16, height: 16, fill: "currentColor", ...t }, /* @__PURE__ */ e.createElement("path", { d: "M12.227 11.52a5.477 5.477 0 0 0 1.246-2.97.5.5 0 0 0-.995-.1 4.478 4.478 0 0 1-.962 2.359l-1.07-1.07C10.794 9.247 11 8.647 11 8V3a3 3 0 0 0-6 0v1.293L1.354.646a.5.5 0 1 0-.708.708l14 14a.5.5 0 0 0 .708-.708zM8 12.5c.683 0 1.33-.152 1.911-.425l.743.743c-.649.359-1.378.59-2.154.66V15h2a.5.5 0 0 1 0 1h-5a.5.5 0 0 1 0-1h2v-1.522a5.502 5.502 0 0 1-4.973-4.929.5.5 0 0 1 .995-.098A4.5 4.5 0 0 0 8 12.5z" }), /* @__PURE__ */ e.createElement("path", { d: "M8.743 10.907 5 7.164V8a3 3 0 0 0 3.743 2.907z" })), at = (t) => /* @__PURE__ */ e.createElement("svg", { xmlns: "http://www.w3.org/2000/svg", width: 16, height: 16, fill: "currentColor", ...t }, /* @__PURE__ */ e.createElement(
  "path",
  {
    fillRule: "evenodd",
    d: "M2.975 8.002a.5.5 0 0 1 .547.449 4.5 4.5 0 0 0 8.956 0 .5.5 0 1 1 .995.098A5.502 5.502 0 0 1 8.5 13.478V15h2a.5.5 0 0 1 0 1h-5a.5.5 0 0 1 0-1h2v-1.522a5.502 5.502 0 0 1-4.973-4.929.5.5 0 0 1 .448-.547z",
    clipRule: "evenodd"
  }
), /* @__PURE__ */ e.createElement("path", { d: "M5 3a3 3 0 1 1 6 0v5a3 3 0 0 1-6 0z" })), nt = (t) => /* @__PURE__ */ e.createElement("svg", { xmlns: "http://www.w3.org/2000/svg", width: 16, height: 16, fill: "currentcolor", ...t }, /* @__PURE__ */ e.createElement("path", { d: "M0 11.5a.5.5 0 0 1 .5-.5h3a.5.5 0 0 1 .5.5v4a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5zm6-5a.5.5 0 0 1 .5-.5h3a.5.5 0 0 1 .5.5v9a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5zm6-6a.5.5 0 0 1 .5-.5h3a.5.5 0 0 1 .5.5v15a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5z" }), /* @__PURE__ */ e.createElement("path", { d: "M0 11.5a.5.5 0 0 1 .5-.5h3a.5.5 0 0 1 .5.5v4a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5zm6-5a.5.5 0 0 1 .5-.5h3a.5.5 0 0 1 .5.5v9a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5zm6-6a.5.5 0 0 1 .5-.5h3a.5.5 0 0 1 .5.5v15a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5z" })), rt = (t) => /* @__PURE__ */ e.createElement("svg", { xmlns: "http://www.w3.org/2000/svg", width: 16, height: 16, fill: "currentcolor", ...t }, /* @__PURE__ */ e.createElement("path", { d: "M0 11.5a.5.5 0 0 1 .5-.5h3a.5.5 0 0 1 .5.5v4a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5zm6-5a.5.5 0 0 1 .5-.5h3a.5.5 0 0 1 .5.5v9a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5z" }), /* @__PURE__ */ e.createElement("path", { d: "M0 11.5a.5.5 0 0 1 .5-.5h3a.5.5 0 0 1 .5.5v4a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5zm6-5a.5.5 0 0 1 .5-.5h3a.5.5 0 0 1 .5.5v9a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5z" }), /* @__PURE__ */ e.createElement("g", { opacity: 0.25 }, /* @__PURE__ */ e.createElement("path", { d: "M12 .5a.5.5 0 0 1 .5-.5h3a.5.5 0 0 1 .5.5v15a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5z" }), /* @__PURE__ */ e.createElement("path", { d: "M12 .5a.5.5 0 0 1 .5-.5h3a.5.5 0 0 1 .5.5v15a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5z" }))), ct = (t) => /* @__PURE__ */ e.createElement("svg", { xmlns: "http://www.w3.org/2000/svg", width: 16, height: 16, fill: "currentcolor", ...t }, /* @__PURE__ */ e.createElement("path", { d: "M0 11.5a.5.5 0 0 1 .5-.5h3a.5.5 0 0 1 .5.5v4a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5z" }), /* @__PURE__ */ e.createElement("path", { d: "M0 11.5a.5.5 0 0 1 .5-.5h3a.5.5 0 0 1 .5.5v4a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5z" }), /* @__PURE__ */ e.createElement("g", { opacity: 0.25 }, /* @__PURE__ */ e.createElement("path", { d: "M6 6.5a.5.5 0 0 1 .5-.5h3a.5.5 0 0 1 .5.5v9a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5z" }), /* @__PURE__ */ e.createElement("path", { d: "M6 6.5a.5.5 0 0 1 .5-.5h3a.5.5 0 0 1 .5.5v9a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5zm6-6a.5.5 0 0 1 .5-.5h3a.5.5 0 0 1 .5.5v15a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5z" }), /* @__PURE__ */ e.createElement("path", { d: "M12 .5a.5.5 0 0 1 .5-.5h3a.5.5 0 0 1 .5.5v15a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5z" }))), lt = (t) => /* @__PURE__ */ e.createElement("svg", { xmlns: "http://www.w3.org/2000/svg", width: 16, height: 16, fill: "currentColor", ...t }, /* @__PURE__ */ e.createElement("g", { opacity: 0.25 }, /* @__PURE__ */ e.createElement("path", { d: "M0 11.5a.5.5 0 0 1 .5-.5h3a.5.5 0 0 1 .5.5v4a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5v-4Zm6-5a.5.5 0 0 1 .5-.5h3a.5.5 0 0 1 .5.5v9a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5v-9Zm6-6a.5.5 0 0 1 .5-.5h3a.5.5 0 0 1 .5.5v15a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5V.5Z" }), /* @__PURE__ */ e.createElement("path", { d: "M0 11.5a.5.5 0 0 1 .5-.5h3a.5.5 0 0 1 .5.5v4a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5v-4Zm6-5a.5.5 0 0 1 .5-.5h3a.5.5 0 0 1 .5.5v9a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5v-9Zm6-6a.5.5 0 0 1 .5-.5h3a.5.5 0 0 1 .5.5v15a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5V.5Z" }))), oe = (t) => /* @__PURE__ */ e.createElement("svg", { xmlns: "http://www.w3.org/2000/svg", width: 20, height: 16, fill: "none", ...t }, /* @__PURE__ */ e.createElement(
  "path",
  {
    fill: "currentColor",
    fillRule: "evenodd",
    d: "M0 2.75A2.75 2.75 0 0 1 2.75 0h14.5A2.75 2.75 0 0 1 20 2.75v10.5A2.75 2.75 0 0 1 17.25 16H2.75A2.75 2.75 0 0 1 0 13.25V2.75ZM2.75 1.5c-.69 0-1.25.56-1.25 1.25v10.5c0 .69.56 1.25 1.25 1.25h14.5c.69 0 1.25-.56 1.25-1.25V2.75c0-.69-.56-1.25-1.25-1.25H2.75Z",
    clipRule: "evenodd"
  }
), /* @__PURE__ */ e.createElement(
  "path",
  {
    fill: "currentColor",
    fillRule: "evenodd",
    d: "M9.47 4.22a.75.75 0 0 1 1.06 0l2.25 2.25a.75.75 0 0 1-1.06 1.06l-.97-.97v4.69a.75.75 0 0 1-1.5 0V6.56l-.97.97a.75.75 0 0 1-1.06-1.06l2.25-2.25Z",
    clipRule: "evenodd"
  }
)), it = (t) => /* @__PURE__ */ e.createElement("svg", { xmlns: "http://www.w3.org/2000/svg", width: 20, height: 16, fill: "none", ...t }, /* @__PURE__ */ e.createElement("g", { fill: "currentColor" }, /* @__PURE__ */ e.createElement("path", { d: "M7.28 4.22a.75.75 0 0 0-1.06 1.06L8.94 8l-2.72 2.72a.75.75 0 1 0 1.06 1.06L10 9.06l2.72 2.72a.75.75 0 1 0 1.06-1.06L11.06 8l2.72-2.72a.75.75 0 0 0-1.06-1.06L10 6.94z" }), /* @__PURE__ */ e.createElement(
  "path",
  {
    fillRule: "evenodd",
    d: "M2.75 0A2.75 2.75 0 0 0 0 2.75v10.5A2.75 2.75 0 0 0 2.75 16h14.5A2.75 2.75 0 0 0 20 13.25V2.75A2.75 2.75 0 0 0 17.25 0zM1.5 2.75c0-.69.56-1.25 1.25-1.25h14.5c.69 0 1.25.56 1.25 1.25v10.5c0 .69-.56 1.25-1.25 1.25H2.75c-.69 0-1.25-.56-1.25-1.25z",
    clipRule: "evenodd"
  }
))), G = (t) => /* @__PURE__ */ e.createElement("svg", { xmlns: "http://www.w3.org/2000/svg", width: 16, height: 16, fill: "none", ...t }, /* @__PURE__ */ e.createElement(
  "path",
  {
    fill: "currentColor",
    fillRule: "evenodd",
    d: "M8 0a.75.75 0 0 1 .75.75v2.5a.75.75 0 0 1-1.5 0V.75A.75.75 0 0 1 8 0Z",
    clipRule: "evenodd"
  }
), /* @__PURE__ */ e.createElement(
  "path",
  {
    fill: "currentColor",
    fillRule: "evenodd",
    d: "M8 12a.75.75 0 0 1 .75.75v2.5a.75.75 0 0 1-1.5 0v-2.5A.75.75 0 0 1 8 12Z",
    clipRule: "evenodd",
    opacity: 0.7
  }
), /* @__PURE__ */ e.createElement(
  "path",
  {
    fill: "currentColor",
    fillRule: "evenodd",
    d: "M12 1.072a.75.75 0 0 1 .274 1.024l-1.25 2.165a.75.75 0 0 1-1.299-.75l1.25-2.165A.75.75 0 0 1 12 1.072Z",
    clipRule: "evenodd"
  }
), /* @__PURE__ */ e.createElement(
  "path",
  {
    fill: "currentColor",
    fillRule: "evenodd",
    d: "M6 11.464a.75.75 0 0 1 .274 1.025l-1.25 2.165a.75.75 0 0 1-1.299-.75l1.25-2.165A.75.75 0 0 1 6 11.464Z",
    clipRule: "evenodd",
    opacity: 0.6
  }
), /* @__PURE__ */ e.createElement(
  "path",
  {
    fill: "currentColor",
    fillRule: "evenodd",
    d: "M14.928 4a.75.75 0 0 1-.274 1.025l-2.165 1.25a.75.75 0 1 1-.75-1.3l2.165-1.25A.75.75 0 0 1 14.928 4Z",
    clipRule: "evenodd"
  }
), /* @__PURE__ */ e.createElement(
  "path",
  {
    fill: "currentColor",
    fillRule: "evenodd",
    d: "M4.536 10a.75.75 0 0 1-.275 1.024l-2.165 1.25a.75.75 0 0 1-.75-1.298l2.165-1.25A.75.75 0 0 1 4.536 10Z",
    clipRule: "evenodd",
    opacity: 0.5
  }
), /* @__PURE__ */ e.createElement(
  "path",
  {
    fill: "currentColor",
    fillRule: "evenodd",
    d: "M16 8a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h2.5A.75.75 0 0 1 16 8Z",
    clipRule: "evenodd"
  }
), /* @__PURE__ */ e.createElement(
  "path",
  {
    fill: "currentColor",
    fillRule: "evenodd",
    d: "M4 8a.75.75 0 0 1-.75.75H.75a.75.75 0 0 1 0-1.5h2.5A.75.75 0 0 1 4 8Z",
    clipRule: "evenodd",
    opacity: 0.4
  }
), /* @__PURE__ */ e.createElement(
  "path",
  {
    fill: "currentColor",
    fillRule: "evenodd",
    d: "M14.928 12a.75.75 0 0 1-1.024.274l-2.165-1.25a.75.75 0 0 1 .75-1.299l2.165 1.25A.75.75 0 0 1 14.928 12Z",
    clipRule: "evenodd",
    opacity: 0.9
  }
), /* @__PURE__ */ e.createElement(
  "path",
  {
    fill: "currentColor",
    fillRule: "evenodd",
    d: "M4.536 6a.75.75 0 0 1-1.025.275l-2.165-1.25a.75.75 0 1 1 .75-1.3l2.165 1.25A.75.75 0 0 1 4.536 6Z",
    clipRule: "evenodd",
    opacity: 0.3
  }
), /* @__PURE__ */ e.createElement(
  "path",
  {
    fill: "currentColor",
    fillRule: "evenodd",
    d: "M12 14.928a.75.75 0 0 1-1.024-.274l-1.25-2.165a.75.75 0 0 1 1.298-.75l1.25 2.165A.75.75 0 0 1 12 14.928Z",
    clipRule: "evenodd",
    opacity: 0.8
  }
), /* @__PURE__ */ e.createElement(
  "path",
  {
    fill: "currentColor",
    fillRule: "evenodd",
    d: "M6 4.536a.75.75 0 0 1-1.024-.275l-1.25-2.165a.75.75 0 1 1 1.299-.75l1.25 2.165A.75.75 0 0 1 6 4.536Z",
    clipRule: "evenodd",
    opacity: 0.2
  }
)), ot = (t) => /* @__PURE__ */ e.createElement("svg", { xmlns: "http://www.w3.org/2000/svg", width: 16, height: 16, fill: "none", ...t }, /* @__PURE__ */ e.createElement("g", { stroke: "currentColor", strokeLinecap: "round", strokeLinejoin: "round", strokeWidth: 1.5 }, /* @__PURE__ */ e.createElement("path", { d: "M13.25 7H9m0 0V2.75M9 7l5.25-5.25M2.75 9H7m0 0v4.25M7 9l-5.25 5.25" }))), st = /* @__PURE__ */ e.forwardRef(
  function({ trackRef: n, ...a }, r) {
    const c = _(), { mergedProps: o, inFocus: s } = we({
      trackRef: n ?? c,
      props: a
    });
    return /* @__PURE__ */ e.createElement(ne.Consumer, null, (l) => l !== void 0 && /* @__PURE__ */ e.createElement("button", { ref: r, ...o }, a.children ? a.children : s ? /* @__PURE__ */ e.createElement(ot, null) : /* @__PURE__ */ e.createElement(Ke, null)));
  }
), oa = /* @__PURE__ */ e.forwardRef(
  function({
    kind: n,
    initialSelection: a,
    onActiveDeviceChange: r,
    onDeviceListChange: c,
    onDeviceSelectError: o,
    exactMatch: s,
    track: l,
    requestPermissions: i,
    onError: u,
    ...d
  }, f) {
    const m = Fe(), E = e.useCallback(
      (w) => {
        m && m.emit(ze.MediaDevicesError, w), u == null || u(w);
      },
      [m, u]
    ), { devices: p, activeDeviceId: h, setActiveMediaDevice: g, className: k } = ke({
      kind: n,
      room: m,
      track: l,
      requestPermissions: i,
      onError: E
    });
    e.useEffect(() => {
      a !== void 0 && g(a);
    }, [g]), e.useEffect(() => {
      typeof c == "function" && c(p);
    }, [c, p]), e.useEffect(() => {
      h && h !== "" && (r == null || r(h));
    }, [h]);
    const C = async (w) => {
      try {
        await g(w, { exact: s });
      } catch (v) {
        if (v instanceof Error)
          o == null || o(v);
        else
          throw v;
      }
    }, b = e.useMemo(
      () => y(d, { className: k }, { className: "lk-list" }),
      [k, d]
    );
    function S(w, v, R) {
      return w === v || R === 0 && v === "default";
    }
    return /* @__PURE__ */ e.createElement("ul", { ref: f, ...b }, p.map((w, v) => /* @__PURE__ */ e.createElement(
      "li",
      {
        key: w.deviceId,
        id: w.deviceId,
        "data-lk-active": S(w.deviceId, h, v),
        "aria-selected": S(w.deviceId, h, v),
        role: "option"
      },
      /* @__PURE__ */ e.createElement("button", { className: "lk-button", onClick: () => C(w.deviceId) }, w.label)
    )));
  }
), sa = /* @__PURE__ */ e.forwardRef(
  function({ label: n = "Allow Audio", ...a }, r) {
    const c = re(), { mergedProps: o } = ee({ room: c, props: a });
    return /* @__PURE__ */ e.createElement("button", { ref: r, ...o }, n);
  }
), ua = /* @__PURE__ */ e.forwardRef(
  function({ label: n, ...a }, r) {
    const c = re(), { mergedProps: o, canPlayAudio: s } = ee({ room: c, props: a }), { mergedProps: l, canPlayVideo: i } = Me({ room: c, props: o }), { style: u, ...d } = l;
    return u.display = s && i ? "none" : "block", /* @__PURE__ */ e.createElement("button", { ref: r, style: u, ...d }, n ?? `Start ${s ? "Video" : "Audio"}`);
  }
);
function se(t, n) {
  switch (t) {
    case M.Source.Microphone:
      return n ? /* @__PURE__ */ e.createElement(at, null) : /* @__PURE__ */ e.createElement(tt, null);
    case M.Source.Camera:
      return n ? /* @__PURE__ */ e.createElement(Je, null) : /* @__PURE__ */ e.createElement(Ye, null);
    case M.Source.ScreenShare:
      return n ? /* @__PURE__ */ e.createElement(it, null) : /* @__PURE__ */ e.createElement(oe, null);
    default:
      return;
  }
}
function ut(t) {
  switch (t) {
    case V.Excellent:
      return /* @__PURE__ */ e.createElement(nt, null);
    case V.Good:
      return /* @__PURE__ */ e.createElement(rt, null);
    case V.Poor:
      return /* @__PURE__ */ e.createElement(ct, null);
    default:
      return /* @__PURE__ */ e.createElement(lt, null);
  }
}
const da = /* @__PURE__ */ e.forwardRef(function({ showIcon: n, ...a }, r) {
  const { buttonProps: c, enabled: o } = Re(a);
  return /* @__PURE__ */ e.createElement("button", { ref: r, ...c }, (n ?? !0) && se(a.source, o), a.children);
}), ue = /* @__PURE__ */ e.forwardRef(function(n, a) {
  const { className: r, quality: c } = ye(n), o = e.useMemo(() => ({ ...y(n, { className: r }), "data-lk-quality": c }), [c, n, r]);
  return /* @__PURE__ */ e.createElement("div", { ref: a, ...o }, n.children ?? ut(c));
}), B = /* @__PURE__ */ e.forwardRef(
  function({ participant: n, ...a }, r) {
    const c = Ze(n), { className: o, infoObserver: s } = e.useMemo(() => He(c), [c]), { identity: l, name: i } = be(s, {
      name: c.name,
      identity: c.identity,
      metadata: c.metadata
    }), u = e.useMemo(() => y(a, { className: o, "data-lk-participant-name": i }), [a, o, i]);
    return /* @__PURE__ */ e.createElement("span", { ref: r, ...u }, i !== "" ? i : l, a.children);
  }
), de = /* @__PURE__ */ e.forwardRef(
  function({ trackRef: n, show: a = "always", ...r }, c) {
    const { className: o, isMuted: s } = Se(n), l = a === "always" || a === "muted" && s || a === "unmuted" && !s, i = e.useMemo(
      () => y(r, {
        className: o
      }),
      [o, r]
    );
    return l ? /* @__PURE__ */ e.createElement("div", { ref: c, ...i, "data-lk-muted": s }, r.children ?? se(n.source, !s)) : null;
  }
), dt = (t) => /* @__PURE__ */ e.createElement(
  "svg",
  {
    width: 320,
    height: 320,
    viewBox: "0 0 320 320",
    preserveAspectRatio: "xMidYMid meet",
    fill: "none",
    xmlns: "http://www.w3.org/2000/svg",
    ...t
  },
  /* @__PURE__ */ e.createElement(
    "path",
    {
      d: "M160 180C204.182 180 240 144.183 240 100C240 55.8172 204.182 20 160 20C115.817 20 79.9997 55.8172 79.9997 100C79.9997 144.183 115.817 180 160 180Z",
      fill: "white",
      fillOpacity: 0.25
    }
  ),
  /* @__PURE__ */ e.createElement(
    "path",
    {
      d: "M97.6542 194.614C103.267 191.818 109.841 192.481 115.519 195.141C129.025 201.466 144.1 205 159.999 205C175.899 205 190.973 201.466 204.48 195.141C210.158 192.481 216.732 191.818 222.345 194.614C262.703 214.719 291.985 253.736 298.591 300.062C300.15 310.997 291.045 320 280 320H39.9997C28.954 320 19.8495 310.997 21.4087 300.062C28.014 253.736 57.2966 214.72 97.6542 194.614Z",
      fill: "white",
      fillOpacity: 0.25
    }
  )
);
function me(t, n = {}) {
  const [a, r] = e.useState(Be(t)), [c, o] = e.useState(a == null ? void 0 : a.isMuted), [s, l] = e.useState(a == null ? void 0 : a.isSubscribed), [i, u] = e.useState(a == null ? void 0 : a.track), [d, f] = e.useState("landscape"), m = e.useRef(), { className: E, trackObserver: p } = e.useMemo(() => je(t), [
    t.participant.sid ?? t.participant.identity,
    t.source,
    z(t) && t.publication.trackSid
  ]);
  return e.useEffect(() => {
    const h = p.subscribe((g) => {
      A.debug("update track", g), r(g), o(g == null ? void 0 : g.isMuted), l(g == null ? void 0 : g.isSubscribed), u(g == null ? void 0 : g.track);
    });
    return () => h == null ? void 0 : h.unsubscribe();
  }, [p]), e.useEffect(() => {
    var h, g;
    return i && (m.current && i.detach(m.current), (h = n.element) != null && h.current && !(ce(t.participant) && (i == null ? void 0 : i.kind) === "audio") && i.attach(n.element.current)), m.current = (g = n.element) == null ? void 0 : g.current, () => {
      m.current && (i == null || i.detach(m.current));
    };
  }, [i, n.element]), e.useEffect(() => {
    var h, g;
    if (typeof ((h = a == null ? void 0 : a.dimensions) == null ? void 0 : h.width) == "number" && typeof ((g = a == null ? void 0 : a.dimensions) == null ? void 0 : g.height) == "number") {
      const k = a.dimensions.width > a.dimensions.height ? "landscape" : "portrait";
      f(k);
    }
  }, [a]), {
    publication: a,
    isMuted: c,
    isSubscribed: s,
    track: i,
    elementProps: y(n.props, {
      className: E,
      "data-lk-local-participant": t.participant.isLocal,
      "data-lk-source": a == null ? void 0 : a.source,
      ...(a == null ? void 0 : a.kind) === "video" && { "data-lk-orientation": d }
    })
  };
}
var mt = "Expected a function", Q = NaN, ft = "[object Symbol]", ht = /^\s+|\s+$/g, gt = /^[-+]0x[0-9a-f]+$/i, vt = /^0b[01]+$/i, Et = /^0o[0-7]+$/i, pt = parseInt, wt = typeof L == "object" && L && L.Object === Object && L, kt = typeof self == "object" && self && self.Object === Object && self, Mt = wt || kt || Function("return this")(), Rt = Object.prototype, yt = Rt.toString, bt = Math.max, St = Math.min, Z = function() {
  return Mt.Date.now();
};
function Ct(t, n, a) {
  var r, c, o, s, l, i, u = 0, d = !1, f = !1, m = !0;
  if (typeof t != "function")
    throw new TypeError(mt);
  n = U(n) || 0, j(a) && (d = !!a.leading, f = "maxWait" in a, o = f ? bt(U(a.maxWait) || 0, n) : o, m = "trailing" in a ? !!a.trailing : m);
  function E(v) {
    var R = r, T = c;
    return r = c = void 0, u = v, s = t.apply(T, R), s;
  }
  function p(v) {
    return u = v, l = setTimeout(k, n), d ? E(v) : s;
  }
  function h(v) {
    var R = v - i, T = v - u, q = n - R;
    return f ? St(q, o - T) : q;
  }
  function g(v) {
    var R = v - i, T = v - u;
    return i === void 0 || R >= n || R < 0 || f && T >= o;
  }
  function k() {
    var v = Z();
    if (g(v))
      return C(v);
    l = setTimeout(k, h(v));
  }
  function C(v) {
    return l = void 0, m && r ? E(v) : (r = c = void 0, s);
  }
  function b() {
    l !== void 0 && clearTimeout(l), u = 0, r = i = c = l = void 0;
  }
  function S() {
    return l === void 0 ? s : C(Z());
  }
  function w() {
    var v = Z(), R = g(v);
    if (r = arguments, c = this, i = v, R) {
      if (l === void 0)
        return p(i);
      if (f)
        return l = setTimeout(k, n), E(i);
    }
    return l === void 0 && (l = setTimeout(k, n)), s;
  }
  return w.cancel = b, w.flush = S, w;
}
function j(t) {
  var n = typeof t;
  return !!t && (n == "object" || n == "function");
}
function It(t) {
  return !!t && typeof t == "object";
}
function Pt(t) {
  return typeof t == "symbol" || It(t) && yt.call(t) == ft;
}
function U(t) {
  if (typeof t == "number")
    return t;
  if (Pt(t))
    return Q;
  if (j(t)) {
    var n = typeof t.valueOf == "function" ? t.valueOf() : t;
    t = j(n) ? n + "" : n;
  }
  if (typeof t != "string")
    return t === 0 ? t : +t;
  t = t.replace(ht, "");
  var a = vt.test(t);
  return a || Et.test(t) ? pt(t.slice(2), a ? 2 : 8) : gt.test(t) ? Q : +t;
}
var Tt = Ct;
const X = /* @__PURE__ */ Oe(Tt);
function xt(t) {
  const n = I(t);
  n.current = t, P(
    () => () => {
      n.current();
    },
    []
  );
}
function At(t, n = 500, a) {
  const r = I();
  xt(() => {
    r.current && r.current.cancel();
  });
  const c = ge(() => {
    const o = X(t, n, a), s = (...l) => o(...l);
    return s.cancel = () => {
      o.cancel();
    }, s.isPending = () => !!r.current, s.flush = () => o.flush(), s;
  }, [t, n, a]);
  return P(() => {
    r.current = X(t, n, a);
  }, [t, n, a]), c;
}
function Nt(t, n, a) {
  const r = (u, d) => u === d, c = t instanceof Function ? t() : t, [o, s] = x(c), l = I(c), i = At(
    s,
    n,
    a
  );
  return r(l.current, c) || (i(c), l.current = c), [o, i];
}
function Lt({
  threshold: t = 0,
  root: n = null,
  rootMargin: a = "0%",
  freezeOnceVisible: r = !1,
  initialIsIntersecting: c = !1,
  onChange: o
} = {}) {
  var s;
  const [l, i] = x(null), [u, d] = x(() => ({
    isIntersecting: c,
    entry: void 0
  })), f = I();
  f.current = o;
  const m = ((s = u.entry) == null ? void 0 : s.isIntersecting) && r;
  P(() => {
    if (!l || !("IntersectionObserver" in window) || m)
      return;
    let h;
    const g = new IntersectionObserver(
      (k) => {
        const C = Array.isArray(g.thresholds) ? g.thresholds : [g.thresholds];
        k.forEach((b) => {
          const S = b.isIntersecting && C.some((w) => b.intersectionRatio >= w);
          d({ isIntersecting: S, entry: b }), f.current && f.current(S, b), S && r && h && (h(), h = void 0);
        });
      },
      { threshold: t, root: n, rootMargin: a }
    );
    return g.observe(l), () => {
      g.disconnect();
    };
  }, [
    l,
    // eslint-disable-next-line react-hooks/exhaustive-deps
    JSON.stringify(t),
    n,
    a,
    m,
    r
  ]);
  const E = I(null);
  P(() => {
    var h;
    !l && ((h = u.entry) != null && h.target) && !r && !m && E.current !== u.entry.target && (E.current = u.entry.target, d({ isIntersecting: c, entry: void 0 }));
  }, [l, u.entry, r, m, c]);
  const p = [
    i,
    !!u.isIntersecting,
    u.entry
  ];
  return p.ref = p[0], p.isIntersecting = p[1], p.entry = p[2], p;
}
const zt = /* @__PURE__ */ e.forwardRef(
  function({
    onTrackClick: n,
    onClick: a,
    onSubscriptionStatusChanged: r,
    trackRef: c,
    manageSubscription: o,
    ...s
  }, l) {
    const i = N(c), u = e.useRef(null);
    e.useImperativeHandle(l, () => u.current);
    const d = Lt({ root: u.current }), [f] = Nt(d, 3e3);
    e.useEffect(() => {
      o && i.publication instanceof H && (f == null ? void 0 : f.isIntersecting) === !1 && (d == null ? void 0 : d.isIntersecting) === !1 && i.publication.setSubscribed(!1);
    }, [f, i, o]), e.useEffect(() => {
      o && i.publication instanceof H && (d == null ? void 0 : d.isIntersecting) === !0 && i.publication.setSubscribed(!0);
    }, [d, i, o]);
    const {
      elementProps: m,
      publication: E,
      isSubscribed: p
    } = me(i, {
      element: u,
      props: s
    });
    e.useEffect(() => {
      r == null || r(!!p);
    }, [p, r]);
    const h = (g) => {
      a == null || a(g), n == null || n({ participant: i == null ? void 0 : i.participant, track: E });
    };
    return /* @__PURE__ */ e.createElement("video", { ref: u, ...m, muted: !0, onClick: h });
  }
), W = /* @__PURE__ */ e.forwardRef(
  function({ trackRef: n, onSubscriptionStatusChanged: a, volume: r, muted: c, ...o }, s) {
    const l = N(n), i = e.useRef(null);
    e.useImperativeHandle(s, () => i.current);
    const {
      elementProps: u,
      isSubscribed: d,
      track: f,
      publication: m
    } = me(l, {
      element: i,
      props: o
    });
    return e.useEffect(() => {
      a == null || a(!!d);
    }, [d, a]), e.useEffect(() => {
      f === void 0 || r === void 0 || (f instanceof Ve ? f.setVolume(r) : A.warn("Volume can only be set on remote audio tracks."));
    }, [r, f]), e.useEffect(() => {
      m === void 0 || c === void 0 || (m instanceof H ? m.setEnabled(!c) : A.warn("Can only call setEnabled on remote track publications."));
    }, [c, m, f]), /* @__PURE__ */ e.createElement("audio", { ref: i, ...u });
  }
);
function Vt(t) {
  const n = !!_e();
  return t.participant && !n ? /* @__PURE__ */ e.createElement(le.Provider, { value: t.participant }, t.children) : /* @__PURE__ */ e.createElement(e.Fragment, null, t.children);
}
function Ft(t) {
  const n = !!_();
  return t.trackRef && !n ? /* @__PURE__ */ e.createElement($.Provider, { value: t.trackRef }, t.children) : /* @__PURE__ */ e.createElement(e.Fragment, null, t.children);
}
const Zt = /* @__PURE__ */ e.forwardRef(
  function({
    trackRef: n,
    children: a,
    onParticipantClick: r,
    disableSpeakingIndicator: c,
    ...o
  }, s) {
    var E, p;
    const l = N(n), { elementProps: i } = te({
      htmlProps: o,
      disableSpeakingIndicator: c,
      onParticipantClick: r,
      trackRef: l
    }), u = Ce(l.participant), d = $e(), f = (E = We()) == null ? void 0 : E.autoSubscription, m = e.useCallback(
      (h) => {
        l.source && !h && d && d.pin.dispatch && qe(l, d.pin.state) && d.pin.dispatch({ msg: "clear_pin" });
      },
      [l, d]
    );
    return /* @__PURE__ */ e.createElement("div", { ref: s, style: { position: "relative" }, ...i }, /* @__PURE__ */ e.createElement(Ft, { trackRef: l }, /* @__PURE__ */ e.createElement(Vt, { participant: l.participant }, a ?? /* @__PURE__ */ e.createElement(e.Fragment, null, z(l) && (((p = l.publication) == null ? void 0 : p.kind) === "video" || l.source === M.Source.Camera || l.source === M.Source.ScreenShare) ? /* @__PURE__ */ e.createElement(
      zt,
      {
        trackRef: l,
        onSubscriptionStatusChanged: m,
        manageSubscription: f
      }
    ) : z(l) && /* @__PURE__ */ e.createElement(
      W,
      {
        trackRef: l,
        onSubscriptionStatusChanged: m
      }
    ), /* @__PURE__ */ e.createElement("div", { className: "lk-participant-placeholder" }, /* @__PURE__ */ e.createElement(dt, null)), /* @__PURE__ */ e.createElement("div", { className: "lk-participant-metadata" }, /* @__PURE__ */ e.createElement("div", { className: "lk-participant-metadata-item" }, l.source === M.Source.Camera ? /* @__PURE__ */ e.createElement(e.Fragment, null, u && /* @__PURE__ */ e.createElement(et, { style: { marginRight: "0.25rem" } }), /* @__PURE__ */ e.createElement(
      de,
      {
        trackRef: {
          participant: l.participant,
          source: M.Source.Microphone
        },
        show: "muted"
      }
    ), /* @__PURE__ */ e.createElement(B, null)) : /* @__PURE__ */ e.createElement(e.Fragment, null, /* @__PURE__ */ e.createElement(oe, { style: { marginRight: "0.25rem" } }), /* @__PURE__ */ e.createElement(B, null, "'s screen"))), /* @__PURE__ */ e.createElement(ue, { className: "lk-participant-metadata-item" }))), /* @__PURE__ */ e.createElement(st, { trackRef: l }))));
  }
);
function ma(t) {
  const n = y(t, { className: "lk-focus-layout" });
  return /* @__PURE__ */ e.createElement("div", { ...n }, t.children);
}
function fa({ trackRef: t, ...n }) {
  return /* @__PURE__ */ e.createElement(Zt, { trackRef: t, ...n });
}
function fe({ tracks: t, ...n }) {
  return /* @__PURE__ */ e.createElement(e.Fragment, null, t.map((a) => /* @__PURE__ */ e.createElement(
    $.Provider,
    {
      value: a,
      key: ie(a)
    },
    O(n.children)
  )));
}
function Ht({
  totalPageCount: t,
  nextPage: n,
  prevPage: a,
  currentPage: r,
  pagesContainer: c
}) {
  const [o, s] = e.useState(!1);
  return e.useEffect(() => {
    let l;
    return c && (l = De(c.current, 2e3).subscribe(
      s
    )), () => {
      l && l.unsubscribe();
    };
  }, [c]), /* @__PURE__ */ e.createElement("div", { className: "lk-pagination-control", "data-lk-user-interaction": o }, /* @__PURE__ */ e.createElement("button", { className: "lk-button", onClick: a }, /* @__PURE__ */ e.createElement(D, null)), /* @__PURE__ */ e.createElement("span", { className: "lk-pagination-count" }, `${r} of ${t}`), /* @__PURE__ */ e.createElement("button", { className: "lk-button", onClick: n }, /* @__PURE__ */ e.createElement(D, null)));
}
const Bt = /* @__PURE__ */ e.forwardRef(
  function({ totalPageCount: n, currentPage: a }, r) {
    const c = new Array(n).fill("").map((o, s) => s + 1 === a ? /* @__PURE__ */ e.createElement("span", { "data-lk-active": !0, key: s }) : /* @__PURE__ */ e.createElement("span", { key: s }));
    return /* @__PURE__ */ e.createElement("div", { ref: r, className: "lk-pagination-indicator" }, c);
  }
);
function ha({ tracks: t, ...n }) {
  const a = e.createRef(), r = e.useMemo(
    () => y(n, { className: "lk-grid-layout" }),
    [n]
  ), { layout: c } = Ie(a, t.length), o = Pe(c.maxTiles, t);
  return Te(a, {
    onLeftSwipe: o.nextPage,
    onRightSwipe: o.prevPage
  }), /* @__PURE__ */ e.createElement("div", { ref: a, "data-lk-pagination": o.totalPageCount > 1, ...r }, /* @__PURE__ */ e.createElement(fe, { tracks: o.tracks }, n.children), t.length > c.maxTiles && /* @__PURE__ */ e.createElement(e.Fragment, null, /* @__PURE__ */ e.createElement(
    Bt,
    {
      totalPageCount: o.totalPageCount,
      currentPage: o.currentPage
    }
  ), /* @__PURE__ */ e.createElement(Ht, { pagesContainer: a, ...o })));
}
const jt = 130, Ot = 140, Y = 1, he = 16 / 10, _t = (1 - he) * -1;
function ga({ tracks: t, orientation: n, ...a }) {
  const r = e.useRef(null), [c, o] = e.useState(0), { width: s, height: l } = xe(r), i = n || (l >= s ? "vertical" : "horizontal"), u = i === "vertical" ? Math.max(s * _t, jt) : Math.max(l * he, Ot), d = Ge(), f = Math.max(i === "vertical" ? (l - d) / u : (s - d) / u, Y);
  let m = Math.round(f);
  Math.abs(f - c) < 0.5 ? m = Math.round(c) : c !== f && o(f);
  const E = Ae(t, m);
  return e.useLayoutEffect(() => {
    r.current && (r.current.dataset.lkOrientation = i, r.current.style.setProperty("--lk-max-visible-tiles", m.toString()));
  }, [m, i]), /* @__PURE__ */ e.createElement("aside", { key: i, className: "lk-carousel", ref: r, ...a }, /* @__PURE__ */ e.createElement(fe, { tracks: E }, a.children));
}
function va({
  value: t,
  onPinChange: n,
  onWidgetChange: a,
  children: r
}) {
  const c = Qe(t);
  return e.useEffect(() => {
    A.debug("PinState Updated", { state: c.pin.state }), n && c.pin.state && n(c.pin.state);
  }, [c.pin.state, n]), e.useEffect(() => {
    A.debug("Widget Updated", { widgetState: c.widget.state }), a && c.widget.state && a(c.widget.state);
  }, [a, c.widget.state]), /* @__PURE__ */ e.createElement(ne.Provider, { value: c }, r);
}
const Ea = /* @__PURE__ */ e.forwardRef(
  function({ trackRef: n, ...a }, r) {
    const d = N(n), f = ae(d, { bands: 7, loPass: 300 });
    return /* @__PURE__ */ e.createElement(
      "svg",
      {
        ref: r,
        width: "100%",
        height: "100%",
        viewBox: "0 0 200 90",
        ...a,
        className: "lk-audio-visualizer"
      },
      /* @__PURE__ */ e.createElement("rect", { x: "0", y: "0", width: "100%", height: "100%" }),
      /* @__PURE__ */ e.createElement(
        "g",
        {
          style: {
            transform: `translate(${(200 - 7 * 10) / 2}px, 0)`
          }
        },
        f.map((m, E) => /* @__PURE__ */ e.createElement(
          "rect",
          {
            key: E,
            x: E * 10,
            y: 90 / 2 - m * 50 / 2,
            width: 6,
            height: m * 50
          }
        ))
      )
    );
  }
);
function pa({ participants: t, ...n }) {
  return /* @__PURE__ */ e.createElement(e.Fragment, null, t.map((a) => /* @__PURE__ */ e.createElement(le.Provider, { value: a, key: a.identity }, O(n.children))));
}
function wa({ volume: t, muted: n }) {
  const a = Ne(
    [M.Source.Microphone, M.Source.ScreenShareAudio, M.Source.Unknown],
    {
      updateOnlyOn: [],
      onlySubscribed: !0
    }
  ).filter((r) => !ce(r.participant) && r.publication.kind === M.Kind.Audio);
  return /* @__PURE__ */ e.createElement("div", { style: { display: "none" } }, a.map((r) => /* @__PURE__ */ e.createElement(
    W,
    {
      key: ie(r),
      trackRef: r,
      volume: t,
      muted: n
    }
  )));
}
const ka = /* @__PURE__ */ e.forwardRef(function({ childrenPosition: n = "before", children: a, ...r }, c) {
  const { name: o } = Le();
  return /* @__PURE__ */ e.createElement("span", { ref: c, ...r }, n === "before" && a, o, n === "after" && a);
});
function $t(t) {
  const n = e.useMemo(() => y(t, { className: "lk-toast" }), [t]);
  return /* @__PURE__ */ e.createElement("div", { ...n }, t.children);
}
const Wt = (t) => {
  const n = [];
  for (let a = 0; a < t; a++)
    n.push([a, t - 1 - a]);
  return n;
}, J = (t) => [[Math.floor(t / 2)], [-1]], qt = (t, n, a) => {
  const [r, c] = x(0), [o, s] = x([[]]);
  P(() => {
    if (t === "thinking")
      s(J(n));
    else if (t === "connecting" || t === "initializing") {
      const i = [...Wt(n)];
      s(i);
    } else s(t === "listening" ? J(n) : t === void 0 ? [new Array(n).fill(0).map((i, u) => u)] : [[]]);
    c(0);
  }, [t, n]);
  const l = I(null);
  return P(() => {
    let i = performance.now();
    const u = (d) => {
      d - i >= a && (c((m) => m + 1), i = d), l.current = requestAnimationFrame(u);
    };
    return l.current = requestAnimationFrame(u), () => {
      l.current !== null && cancelAnimationFrame(l.current);
    };
  }, [a, n, t, o.length]), o[r % o.length];
}, Dt = /* @__PURE__ */ new Map([
  ["connecting", 2e3],
  ["initializing", 2e3],
  ["listening", 500],
  ["thinking", 150]
]), Gt = (t, n) => {
  if (t === void 0)
    return 1e3;
  let a = Dt.get(t);
  if (a)
    switch (t) {
      case "connecting":
        a /= n;
        break;
    }
  return a;
}, Qt = /* @__PURE__ */ e.forwardRef(
  function({ state: n, options: a, barCount: r = 15, trackRef: c, children: o, ...s }, l) {
    const i = y(s, { className: "lk-audio-bar-visualizer" });
    let u = _();
    c && (u = c);
    const d = ae(u, {
      bands: r,
      loPass: 100,
      hiPass: 200
    }), f = (a == null ? void 0 : a.minHeight) ?? 20, m = (a == null ? void 0 : a.maxHeight) ?? 100, E = qt(
      n,
      r,
      Gt(n, r) ?? 100
    );
    return /* @__PURE__ */ e.createElement("div", { ref: l, ...i, "data-lk-va-state": n }, d.map(
      (p, h) => o ? O(o, {
        "data-lk-highlighted": E.includes(h),
        "data-lk-bar-index": h,
        className: "lk-audio-bar",
        style: { height: `${Math.min(m, Math.max(f, p * 100 + 5))}%` }
      }) : /* @__PURE__ */ e.createElement(
        "span",
        {
          key: h,
          "data-lk-highlighted": E.includes(h),
          "data-lk-bar-index": h,
          className: `lk-audio-bar ${E.includes(h) && "lk-highlighted"}`,
          style: {
            // TODO transform animations would be more performant, however the border-radius gets distorted when using scale transforms. a 9-slice approach (or 3 in this case) could work
            // transform: `scale(1, ${Math.min(maxHeight, Math.max(minHeight, volume))}`,
            height: `${Math.min(m, Math.max(f, p * 100 + 5))}%`
          }
        }
      )
    ));
  }
), Ma = /* @__PURE__ */ e.forwardRef(
  function({
    children: n,
    disableSpeakingIndicator: a,
    onParticipantClick: r,
    trackRef: c,
    ...o
  }, s) {
    const l = N(c), { elementProps: i } = te({
      trackRef: l,
      htmlProps: o,
      disableSpeakingIndicator: a,
      onParticipantClick: r
    });
    return /* @__PURE__ */ e.createElement("div", { ref: s, style: { position: "relative", minHeight: "160px" }, ...i }, /* @__PURE__ */ e.createElement($.Provider, { value: l }, n ?? /* @__PURE__ */ e.createElement(e.Fragment, null, z(l) && /* @__PURE__ */ e.createElement(W, { trackRef: l }), /* @__PURE__ */ e.createElement(Qt, { barCount: 7, options: { minHeight: 8 } }), /* @__PURE__ */ e.createElement("div", { className: "lk-participant-metadata" }, /* @__PURE__ */ e.createElement("div", { className: "lk-participant-metadata-item" }, /* @__PURE__ */ e.createElement(de, { trackRef: l }), /* @__PURE__ */ e.createElement(B, null)), /* @__PURE__ */ e.createElement(ue, { className: "lk-participant-metadata-item" })))));
  }
);
function Ra(t) {
  const [n, a] = e.useState(void 0), r = K(t.room);
  return e.useEffect(() => {
    switch (r) {
      case F.Reconnecting:
        a(
          /* @__PURE__ */ e.createElement(e.Fragment, null, /* @__PURE__ */ e.createElement(G, { className: "lk-spinner" }), " Reconnecting")
        );
        break;
      case F.Connecting:
        a(
          /* @__PURE__ */ e.createElement(e.Fragment, null, /* @__PURE__ */ e.createElement(G, { className: "lk-spinner" }), " Connecting")
        );
        break;
      case F.Disconnected:
        a(/* @__PURE__ */ e.createElement(e.Fragment, null, "Disconnected"));
        break;
      default:
        a(void 0);
        break;
    }
  }, [r]), n ? /* @__PURE__ */ e.createElement($t, { className: "lk-toast-connection-state" }, n) : /* @__PURE__ */ e.createElement(e.Fragment, null);
}
const ya = /* @__PURE__ */ e.forwardRef(
  function({ entry: n, hideName: a = !1, hideTimestamp: r = !1, messageFormatter: c, ...o }, s) {
    var f, m, E;
    const l = e.useMemo(() => c ? c(n.message) : n.message, [n.message, c]), i = !!n.editTimestamp, u = new Date(n.timestamp), d = navigator ? navigator.language : "en-US";
    return /* @__PURE__ */ e.createElement(
      "li",
      {
        ref: s,
        className: "lk-chat-entry",
        title: u.toLocaleTimeString(d, { timeStyle: "full" }),
        "data-lk-message-origin": (f = n.from) != null && f.isLocal ? "local" : "remote",
        ...o
      },
      (!r || !a || i) && /* @__PURE__ */ e.createElement("span", { className: "lk-meta-data" }, !a && /* @__PURE__ */ e.createElement("strong", { className: "lk-participant-name" }, ((m = n.from) == null ? void 0 : m.name) ?? ((E = n.from) == null ? void 0 : E.identity)), (!r || i) && /* @__PURE__ */ e.createElement("span", { className: "lk-timestamp" }, i && "edited ", u.toLocaleTimeString(d, { timeStyle: "short" }))),
      /* @__PURE__ */ e.createElement("span", { className: "lk-message-body" }, l)
    );
  }
);
function ba(t) {
  return Ue(t, Xe()).map((n, a) => {
    if (typeof n == "string")
      return n;
    {
      const r = n.content.toString(), c = n.type === "url" ? /^http(s?):\/\//.test(r) ? r : `https://${r}` : `mailto:${r}`;
      return /* @__PURE__ */ e.createElement("a", { className: "lk-chat-link", key: a, href: c, target: "_blank", rel: "noreferrer" }, r);
    }
  });
}
export {
  Ea as A,
  Qt as B,
  aa as C,
  na as D,
  Je as E,
  ma as F,
  ha as G,
  D as H,
  Ke as I,
  et as J,
  tt as K,
  va as L,
  oa as M,
  at as N,
  nt as O,
  Zt as P,
  rt as Q,
  wa as R,
  ra as S,
  da as T,
  ct as U,
  zt as V,
  lt as W,
  oe as X,
  it as Y,
  G as Z,
  ot as _,
  ya as a,
  dt as b,
  ca as c,
  la as d,
  ia as e,
  ua as f,
  ga as g,
  fa as h,
  Ra as i,
  fe as j,
  Ma as k,
  $t as l,
  ba as m,
  ea as n,
  ta as o,
  st as p,
  sa as q,
  ue as r,
  W as s,
  B as t,
  de as u,
  pa as v,
  ka as w,
  Vt as x,
  Ft as y,
  Ye as z
};
